warp10/src/main/java/io/warp10/script/functions/SNAPSHOT.java

```diff
@@ -442,26 +440,16 @@ private static String processString(String s) {
     while(idx < chars.length) {
       if ('\'' == chars[idx] || chars[idx] < ' ') {
 
-        if (null == sb) {
-          sb = new StringBuilder(s.length() + 2);
-        }
-
         sb.append(chars, lastIdx, idx - lastIdx);
-        sb.append("%" + Integer.toHexString((int) chars[idx]));
+        sb.append("%" + (chars[idx] >>> 4) + (chars[idx] & 0xF));
```

Comment:
You need to convert the shifted / masked value to an hex digit. You implementation works for the first digit as chars[idx] is < 0x20 but fails for chars[idx] & 0xF as the value can go from 0 to 0xF which will appear as a 2 digit number.